docs/GettingStarted.md

```diff
@@ -1,19 +1,22 @@
+Apache Spark is a general purpose parallel computational engine for analytics at scale. At its core, it has a batch design center and can access disparate data sources in a highly parallelized manner for its distributed computations. Typically, data is fetched lazily as a result of SQL query or a Dataset (RDD) getting materialized. This can be quite inefficient and expensive since most workload require the data set to be repeatedly processed. Caching within Spark is immutable and requires the application to periodically refresh the data set, let alone having to bear the burden of duplicating the dataset.
 
-Analytic processing requires massive data sets to be repeatedly copied and data to be reformatted to suit Spark. In many cases, it ultimately fails to deliver the promise of interactive analytic performance. For instance, each time an aggregation is run on a large Cassandra table, it necessitates streaming the entire table into Spark to do the aggregation. Caching within Spark is immutable and results in stale insight.
+Analytic processing requires large datasets to be repeatedly copied from external datasource like HDFS, into Spark. Copying data, reformatting it (into columnar format, depending on where the data is being copied from) and moving it across process and machine boundaries can be very expensive. As a result, we see that in several cases, applications using Spark with an external data source fail to deliver the promise of interactive analytic performance. For instance, each time an aggregation is run on a large Cassandra table, it necessitates streaming the entire table into Spark to do the aggregation. The alternative to working with a stateful store is to cache the data in Spark. This of course suffers from the problems associated with stale data.
 
 ## The SnappyData Approach
-At SnappyData, a very different approach is taken. SnappyData fuses a low latency, highly available in-memory transactional database (GemFireXD) into Spark with shared memory management and optimizations. Data in the highly available in-memory store is laid out using the same columnar format as Spark. All query engine operators are more optimized through better vectorization and code generation. The net effect is, an order of magnitude performance improvement when compared to native Spark caching, and more than two orders of magnitude better Spark performance when working with external data sources.
+SnappyData fuses a low latency, highly available in-memory transactional database (GemFire) into Spark with shared memory management and optimizations. Data in the highly available in-memory store is laid out using the same columnar format as Spark. Query engine operators are optimized through better vectorization and code generation. The net effect of these changes are, an order of magnitude performance improvement when compared to native Spark caching, and more than two orders of magnitude better Spark performance when working with external data sources.
 
```

Comment:
"using the same columnar format as Spark" => "using a custom columnar format somewhat similar to the layout used by Spark caching"